### PR TITLE
chore: upgrade storybook + vite builder

### PR DESCRIPTION
**Pull request by g10**
## Description

This upgrades Storybook to V7 and changes the visual testing tool to Chromatic (Storybook only).
~~Thing to test afterward: Check if TurboSnap is enabled. Turbosnap is an optimization tool from Chromatic that compares only changed snapshots.~~

**Fixes** # 18805

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

NO QA

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by paloth — DISMISSED** [review]

File: centreon/packages/ui/tsconfig.json [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
-    "./src/**/*.test.tsx",
-    "./src/**/*.stories.tsx",
-    "../../node_modules",
-  ]
+//    "./src/**/*.test.ts",
+//    "./src/**/*.test.tsx",
+//    "./src/**/*.stories.tsx",
+    "../../node_modules"
+  ],
+//  "references": [{ "path": "./tsconfig.node.json" }]
```
Comment (Thebarda) [inline centreon/packages/ui/tsconfig.json:28]:
```suggestion
    "../../node_modules"
  ],

```

File: centreon/webpack.config.dev.js [hunk 2]
```diff
@@ -77,6 +77,9 @@ module.exports = merge(
     plugins: devServerPlugins,
     resolve: {
       alias: {
+        '@centreon/ui/fonts': path.resolve(
+          './node_modules/@centreon/ui/public/fonts'
+        ),
```
Comment (Thebarda) [inline centreon/webpack.config.dev.js:82]:
```suggestion
```

Comment (Thebarda) [inline centreon/webpack.config.dev.js:82]:
```suggestion
```

Comment (g10) [reply centreon/webpack.config.dev.js]: does it work without? afaik. webpack needs to be configured to handle the alias

**Comment by g10** [comment]
LGTM

**Review by paloth — APPROVED** [review]

**Pull request merged**